### ISSUE TITLE
resetForm doesn't rerun mapPropsToValues as said in docs

### ISSUE THREAD

**Issue by bostrom** [jaredpalmer/formik#1360]
## 📖 Documentation

As per the Formik documentation, it should be possible to pass new values to `resetForm` and those values would be passed to `mapPropsToValues`, the result of which would populate the form:

> Imperatively reset the form. This will clear errors and touched, set isSubmitting to false, isValidating to false, and rerun mapPropsToValues with the current WrappedComponent's props or what's passed as an argument.

However, I found that `mapPropsToValues` isn't rerun when calling `resetForm`, instead the `resetForm` method expects the `values` argument to be in the format that `mapPropsToValues` would output. 

I.e. instead of doing

```
const newValues = async sendUpdateToServer(values);
resetForm(newValues);
```

one would have to do

```
const newValues = async sendUpdateToServer(values);
resetForm(mapPropsToValues(newValues));
```

where `mapPropsToValues` is the same map function passed to `Formik` or `withFormik`.

**Comment by marioharper** [comment]
Yeah, this documentation seems misleading. It doesn't look to me like `resetForm` even has access to the `mapPropsToValues` method that is provided to the HoC. https://github.com/jaredpalmer/formik/blob/c5243947fcf21cd873dd86b57689a30ffc46eab8/src/Formik.tsx#L531

Maybe the docs should be changed to say something along the lines of:

> Imperatively reset the form. This will clear errors and touched, set isSubmitting to false, isValidating to false, and rerun mapPropsToValues with the current WrappedComponent's props. If provided, `nextValues` will be used as the forms new `initialValues` (useful for calling resetForm within componentWillReceiveProps).

**Comment by robfr77** [comment]
Yes, if one imperatively sets a value after the mapPropsToValues init, how does one later resetForm to the mapPropsToValues initial state?
